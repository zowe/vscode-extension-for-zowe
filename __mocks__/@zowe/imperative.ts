/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * This interface defines the options that can be sent into the dwanload data set function
 */

const log4js = require("log4js");

export interface ICliOptions {
    profileRootDirectory: string;
    type: string;
}
export interface IConfigLogging {
    "log4jsConfig"?: any;
}

export interface ILoadOptions {
    name?: string;
    loadDefault?: boolean;
}
export interface ISessionOptions {
    user: string;
    password: string;
    hostname: string;
    port: number;
    protocol: string;
    type: string;
}


export class BrightProfile {
    constructor(public profile: Profile) { }
}

// tslint:disable-next-line:max-classes-per-file
export class Session {
    constructor(public ISession: ISessionOptions) { }
}

// tslint:disable-next-line:class-name
// tslint:disable-next-line:max-classes-per-file
export class Profile {
    constructor(public name: string, public type: string) { }
}

// tslint:disable-next-line:max-classes-per-file
export class CliProfileManager {
    // tslint:disable-next-line:no-empty
    constructor(options: ICliOptions) { }

    public load(opts: ILoadOptions) {
        return { name: "profile1", profile: {}, type: "zosmf" };
    }
    public loadAll() {
        return [
            { name: "sestest", profile: {}, type: "zosmf" },
            { name: "profile1", profile: {}, type: "zosmf" },
            { name: "profile2", profile: {}, type: "zosmf" }];
    }
    public getAllProfileNames(){
        return ["name1", "name2"];
    }
    public save(){
        return { name: "profile1", profile: {}, type: "zosmf" };
    }
    public update(){
        return { name: "profile1", profile: {}, type: "zosmf" };
    }
    public delete(){
        return { name: "profile1", profile: {}, type: "zosmf" };
    }
    public get configurations() {
        return [{
            type: "zosmf",
            schema: {
                type: "object",
                title: "test profile",
                description: "test profile",
                properties: {
                    sum: {
                        type: "number"
                         }
                },
                required: ["sum"]
            },
        },
        {
            type: "banana",
            schema: {
                type: "object",
                title: "test banana",
                description: "test banana",
                properties: {
                    sum: {
                        type: "number"
                         }
                },
                required: ["sum"]
            },
        }];
    }
}

// tslint:disable-next-line:max-classes-per-file
export class ImperativeConfig {
    public static instance = {
        cliHome: "./__tests__/.zowe",
        loadedConfig: {}
    };
    public loadedConfig = {
        defaultHome: "./__tests__/.zowe",
        envVariablePrefix: "ZOWE"
    };
    public cliHome: "./__tests__/.zowe";

}

// tslint:disable-next-line:max-classes-per-file
export class CredentialManagerFactory {
}

// tslint:disable-next-line:max-classes-per-file
export class DefaultCredentialManager {
     public test: "test";
}

// tslint:disable-next-line:max-classes-per-file
export class AbstractCredentialManager {
}

// tslint:disable-next-line:max-classes-per-file
export class Logger {
    public static initLogger(loggingConfig: IConfigLogging): any {
    }
    public static getAppLogger(): Logger {
        return log4js.getLogger("app");
    }
}

// tslint:disable-next-line:max-classes-per-file
export class TextUtils {
    public static prettyJson(object: any, options?: any, color?: boolean, append?: string): string {
        return JSON.stringify(object);
    }
}
