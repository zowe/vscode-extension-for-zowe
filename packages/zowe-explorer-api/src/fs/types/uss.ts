/**
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import * as vscode from "vscode";
import { ConflictData, DirEntry, EntryMetadata, FileEntry } from ".";

export class UssFile extends FileEntry {
    public name: string;
    public metadata: EntryMetadata;
    public type: vscode.FileType;
    public wasAccessed: boolean;

    public ctime: number;
    public mtime: number;
    public size: number;
    public conflictData?: ConflictData;
    public data: Uint8Array;
    public etag?: string;
    public permissions?: vscode.FilePermission;

    public constructor(name: string) {
        super(name);
    }
}

export class UssDirectory extends DirEntry {
    public constructor(name?: string) {
        super(name ?? "");
    }
}
