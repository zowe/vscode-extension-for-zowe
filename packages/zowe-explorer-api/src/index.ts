/**
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

export * from "./extend";
export * from "./globals/Constants";
export * from "./globals/Gui";
export * from "./globals/GuiOptions";
export * from "./logger";
export * from "./PersistentFilters";
export * from "./profiles";
export * from "./security";
export * from "./tree";
export * from "./utils";
export * as contextually from "./shared/context";
export * from "./vscode";
export * from "./Types";
export * as imperative from "@zowe/imperative";
