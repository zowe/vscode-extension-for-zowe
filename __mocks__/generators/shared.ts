/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import * as imperative from "@zowe/imperative";
import * as zowe from "@zowe/cli";
import { ValidProfileEnum } from "../../src/Profiles";

export function generateSession() {
    return new imperative.Session({
        user: "fake",
        password: "fake",
        hostname: "fake",
        protocol: "https",
        type: "basic",
    });
}

export function generateSessionNoCredentials() {
    return new imperative.Session({
        user: "",
        password: "",
        hostname: "fake",
        port: 443,
        protocol: "https",
        type: "basic",
    });
}

export function generateProfile(): imperative.IProfileLoaded {
    return {
        name: "sestest",
        profile: {
            user: undefined,
            password: undefined
        },
        type: "zosmf",
        message: "",
        failNotFound: false
    };
}

export function generateFileResponse(): zowe.IZosFilesResponse {
    return {
        success: true,
        commandResponse: null,
        apiResponse: {
            etag: "123"
        }
    };
};

export function generateTreeView() {
    return {
        reveal: jest.fn(),
        onDidExpandElement: jest.fn(),
        onDidCollapseElement: jest.fn(),
        selection: [],
        onDidChangeSelection: jest.fn(),
        visible: true,
        onDidChangeVisibility: jest.fn()
    };
}

export function generateInstanceOfProfile(profile: imperative.IProfileLoaded) {
    return {
        allProfiles: [{name: "firstName"}, {name: "secondName"}],
        defaultProfile: {name: "firstName"},
        getDefaultProfile: jest.fn(),
        promptCredentials: jest.fn(),
        loadNamedProfile: jest.fn(),
        usesSecurity: true,
        validProfile: ValidProfileEnum.VALID,
        checkCurrentProfile: jest.fn(),
        getProfiles: jest.fn(() => {
            return [{name: profile.name, profile}, {name: profile.name, profile}];
        }),
        refresh: jest.fn()
    };
}