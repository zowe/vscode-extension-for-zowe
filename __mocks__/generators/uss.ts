/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ZoweUSSNode } from "../../src/uss/ZoweUSSNode";
import * as vscode from "vscode";
import * as globals from "../../src/globals";
import { getIconByNode } from "../../src/generators/icons";

const mockUSSRefresh = jest.fn();
export function generateUSSTree(favoriteNodes: ZoweUSSNode[], sessionNodes: ZoweUSSNode[], treeView: any): any {
    return {
        mSessionNodes: [...sessionNodes],
        mFavorites: favoriteNodes,
        addSession: jest.fn(),
        refresh: mockUSSRefresh,
        refreshAll: mockUSSRefresh,
        removeRecall: jest.fn(),
        addRecall: jest.fn(),
        treeView,
        checkCurrentProfile: jest.fn(),
        refreshElement: jest.fn(),
        getChildren: jest.fn(),
        addFavorite: jest.fn(),
        removeFavorite: jest.fn(),
        initializeUSSFavorites: jest.fn(),
        searchInLoadedItems: jest.fn(),
        getTreeView: jest.fn().mockImplementation(() => treeView),
        setItem: jest.fn(),
        addHistory: jest.fn()
    };
}

export function generateUSSNode(session, profile) {
    const mParent = new ZoweUSSNode("parentNode", vscode.TreeItemCollapsibleState.Expanded, null, session, null, false, profile.name);
    const ussNode = new ZoweUSSNode("usstest", vscode.TreeItemCollapsibleState.Expanded, mParent, session, null, false, profile.name);
    ussNode.contextValue = globals.USS_SESSION_CONTEXT;
    ussNode.fullPath = "/u/myuser";
    return ussNode;
}

export function generateFavoriteUSSNode(session, profile) {
    const ussNodeF = new ZoweUSSNode("[profile]: usstest", vscode.TreeItemCollapsibleState.Expanded, null, session, null, false, profile.name);
    const mParent = new ZoweUSSNode("Favorites", vscode.TreeItemCollapsibleState.Expanded, null, session, null, false, profile.name);
    mParent.contextValue = globals.FAVORITE_CONTEXT;
    ussNodeF.contextValue = globals.DS_TEXT_FILE_CONTEXT + globals.FAV_SUFFIX;
    ussNodeF.fullPath = "/u/myuser/usstest";
    ussNodeF.tooltip = "/u/myuser/usstest";
    return ussNodeF;
}

export function addSessionNode(theTree, theSession, theProfile) {
    const newSessNode = new ZoweUSSNode("ussTestSess", vscode.TreeItemCollapsibleState.Collapsed, null, theSession, null, false, theProfile.name)
    theTree.mSessionNodes.push(newSessNode);
    const sessionIndex = theTree.mSessionNodes.length - 1;
    theTree.mSessionNodes[sessionIndex].contextValue = globals.USS_SESSION_CONTEXT;
    theTree.mSessionNodes[sessionIndex].fullPath = "test";
    const targetIcon = getIconByNode(theTree.mSessionNodes[sessionIndex]);
    if (targetIcon) {
        theTree.mSessionNodes[1].iconPath = targetIcon.path;
    }
    return theTree;
}