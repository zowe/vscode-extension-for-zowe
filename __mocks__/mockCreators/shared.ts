/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import * as imperative from "@zowe/imperative";
import { ZoweTreeProvider } from "../../src/abstract/ZoweTreeProvider";
import { ZoweDatasetNode } from "../../src/dataset/ZoweDatasetNode";
import { ZoweUSSNode } from "../../src/uss/ZoweUSSNode";
import * as vscode from "vscode";
import { ValidProfileEnum } from "../../src/Profiles";
import * as utils from "../../src/utils";
import * as zowe from "@zowe/cli";

export function createPersistentConfig() {
    return {
        persistence: true,
        get: () => {
            return {
                sessions: ["sestest", "profile1", "profile2"],
                favorites: ["[sestest]: TEST.PDS", "[profile1]: /u/myuser.txt{textFile}", "[profile2]: /u/myuser"]
            };
        },
        update: jest.fn(()=>{
            return {};
        })
    };
}

export function createISession() {
    return new imperative.Session({
        user: "fake",
        password: "fake",
        hostname: "fake",
        port: 1443,
        protocol: "https",
        type: "basic",
    });
}

export function createISessionWithoutCredentials() {
    return new imperative.Session({
        user: "",
        password: "",
        hostname: "fake",
        protocol: "https",
        type: "basic",
    });
}

export function createBasicZosmfSession(profile: imperative.IProfileLoaded) {
    return zowe.ZosmfSession.createBasicZosmfSession(profile.profile);
}

export function removeNodeFromArray(badNode, array) {
    array.splice(array.findIndex(
        (nodeInArray) => badNode.getProfileName() === nodeInArray.getProfileName()
    ), 1);
}

export function createIProfile(): imperative.IProfileLoaded {
    return {
        name: "sestest",
        profile: {
            user: undefined,
            password: undefined
        },
        type: "zosmf",
        message: "",
        failNotFound: false
    };
}

export function createInvalidIProfile(): imperative.IProfileLoaded {
    return {
        name: "sestest",
        profile: {
            type: "zosmf",
            host: null,
            port: 1443,
            user: null,
            password: null,
            rejectUnauthorized: false,
            name: "testName"
        },
        type: "zosmf",
        message: "",
        failNotFound: false
    };
}

export function createValidIProfile(): imperative.IProfileLoaded {
    return {
        name: "sestest",
        profile: {
            type : "zosmf",
            host: "test",
            port: 1443,
            user: "test",
            password: "test",
            rejectUnauthorized: false,
            name: "testName"
        },
        type: "zosmf",
        message: "",
        failNotFound: false
    };
}

export function createTreeView(): vscode.TreeView<ZoweTreeProvider> {
    return {
        reveal: jest.fn(),
        onDidExpandElement: jest.fn(),
        onDidCollapseElement: jest.fn(),
        selection: [],
        onDidChangeSelection: jest.fn(),
        visible: true,
        onDidChangeVisibility: jest.fn(),
        dispose: jest.fn()
    };
}

export function createTextDocument(name: string, sessionNode?: ZoweDatasetNode | ZoweUSSNode): vscode.TextDocument {
    return {
        fileName: sessionNode ? `/${sessionNode.label}/${name}` : name,
        uri: null,
        isUntitled: null,
        languageId: null,
        version: null,
        isDirty: null,
        isClosed: null,
        save: null,
        eol: null,
        lineCount: null,
        lineAt: null,
        offsetAt: null,
        positionAt: null,
        getText: jest.fn(),
        getWordRangeAtPosition: null,
        validateRange: null,
        validatePosition: null
    };
}

export function createInstanceOfProfile(profile: imperative.IProfileLoaded) {
    return {
        allProfiles: [{ name: "sestest" }, { name: "profile1" }, { name: "profile2" }],
        defaultProfile: { name: "profile1" },
        getDefaultProfile: jest.fn(),
        promptCredentials: jest.fn(),
        loadNamedProfile: jest.fn(),
        usesSecurity: true,
        validProfile: ValidProfileEnum.VALID,
        checkCurrentProfile: jest.fn(() => {
            return {status: "active", name: "fake"};
        }),
        profilesForValidation: [{status: "active", name: "fake"}],
        validateProfiles: jest.fn(),
        editSession: jest.fn(),
        createNewConnection: jest.fn(() => {
            return { newprofile: "fake" };
        }),
        getProfiles: jest.fn(() => {
            return [{ name: profile.name, profile }, { name: profile.name, profile }];
        }),
        refresh: jest.fn(),
    } as any;
}

export function createFileResponse(theResponse) {
    return {
        success: true,
        commandResponse: "",
        apiResponse: theResponse
    } as any;
}

export function createQuickPickItem(): vscode.QuickPickItem {
    return new utils.FilterDescriptor("\uFF0B " + "Create a new filter");
}

export function createQuickPickContent(entered: any, item: vscode.QuickPickItem): any {
    return {
        placeholder: "Choose \"Create new...\" to define a new profile or select an existing profile to Add to the Data Set Explorer",
        activeItems: [item],
        ignoreFocusOut: true,
        items: [item],
        value: entered,
        show: jest.fn(),
        hide: jest.fn(),
        onDidAccept: jest.fn(),
        onDidChangeValue: jest.fn(),
        dispose: jest.fn()
    };
}

export function createInputBox(value: string): any {
    const inputBox: vscode.InputBox = {
        value,
        title: null,
        enabled: true,
        busy: false,
        show: jest.fn(),
        hide: jest.fn(),
        step: null,
        dispose: jest.fn(),
        ignoreFocusOut: false,
        totalSteps: null,
        placeholder: undefined,
        password: false,
        onDidChangeValue: jest.fn(),
        onDidAccept: jest.fn(),
        onDidHide: jest.fn(),
        buttons: [],
        onDidTriggerButton: jest.fn(),
        prompt: undefined,
        validationMessage: undefined
    };
    return inputBox;
}

export function createWorkspaceConfiguration(): vscode.WorkspaceConfiguration {
    return {
        get: jest.fn(),
        update: jest.fn(),
        has: jest.fn(),
        inspect: jest.fn()
    };
}
