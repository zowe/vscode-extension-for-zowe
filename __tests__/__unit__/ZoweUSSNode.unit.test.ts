/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

// tslint:disable:no-shadowed-variable
jest.mock("vscode");
jest.mock("@zowe/imperative");
jest.mock("Session");
import { Session, IProfileLoaded, Logger } from "@zowe/imperative";
import * as vscode from "vscode";
import { ZoweUSSNode } from "../../src/uss/ZoweUSSNode";
import * as sharedUtils from "../../src/shared/utils";
import { Profiles } from "../../src/Profiles";
import { DS_TEXT_FILE_CONTEXT, DS_BINARY_FILE_CONTEXT, FAV_SUFFIX,
    FAVORITE_CONTEXT, USS_SESSION_CONTEXT, USS_DIR_CONTEXT, DS_PDS_CONTEXT } from "../../src/globals";

describe("Unit Tests (Jest)", () => {
    Object.defineProperty(vscode.commands, "executeCommand", {value: jest.fn()});
    // Globals
    const session = new Session({
        user: "fake",
        password: "fake",
        hostname: "fake",
        protocol: "https",
        type: "basic",
    });
    const profileOne: IProfileLoaded = {
        name: "aProfile",
        profile: {},
        type: "zosmf",
        message: "",
        failNotFound: false
    };

    const showErrorMessage = jest.fn();
    Object.defineProperty(vscode.window, "showErrorMessage", {value: showErrorMessage});

    const ProgressLocation = jest.fn().mockImplementation(() => {
        return {
            Notification: 15
        };
    });

    const withProgress = jest.fn().mockImplementation((progLocation, callback) => {
        return callback();
    });

    Object.defineProperty(vscode, "ProgressLocation", {value: ProgressLocation});
    Object.defineProperty(vscode.window, "withProgress", {value: withProgress});
    beforeEach(() => {
        const mockLoadNamedProfile = jest.fn();
        mockLoadNamedProfile.mockReturnValue(profileOne);
        Object.defineProperty(Profiles, "getInstance", {
            value: jest.fn(() => {
                return {
                    allProfiles: [{name: "firstName"}, {name: "secondName"}],
                    getDefaultProfile: {name: "firstName"},
                    loadNamedProfile: mockLoadNamedProfile,
                    refresh: jest.fn()
                };
            })
        });
        withProgress.mockImplementation((progLocation, callback) => {
            return callback();
        });
    });
    afterEach(() => {
        jest.resetAllMocks();
    });

    Profiles.createInstance(Logger.getAppLogger());

    /*************************************************************************************************************
     * Checks that the ZoweUSSNode structure is the same as the snapshot
     *************************************************************************************************************/
    it("Checks that the ZoweUSSNode structure matches the snapshot", async () => {
        const rootNode = new ZoweUSSNode(
            "root", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, undefined);
        rootNode.contextValue = USS_SESSION_CONTEXT;
        rootNode.dirty = true;
        const testDir = new ZoweUSSNode(
            "testDir", vscode.TreeItemCollapsibleState.Collapsed, rootNode, null, null, false, profileOne.name, undefined);
        const testFile = new ZoweUSSNode(
            "testFile", vscode.TreeItemCollapsibleState.None, testDir, null, null, false, profileOne.name, undefined);
        testFile.contextValue = DS_TEXT_FILE_CONTEXT;
        expect(JSON.stringify(rootNode.iconPath)).toContain("folder-closed.svg");
        expect(JSON.stringify(testDir.iconPath)).toContain("folder-closed.svg");
        expect(JSON.stringify(testFile.iconPath)).toContain("document.svg");
        rootNode.iconPath = "Ref: 'folder.svg'";
        testDir.iconPath = "Ref: 'folder.svg'";
        testFile.iconPath = "Ref: 'document.svg'";
        expect(testFile).toMatchSnapshot();
    });

    /*************************************************************************************************************
     * Creates a ZoweUSSNode and checks that its childs are all initialized by the constructor
     *************************************************************************************************************/
    it("Testing that the ZoweUSSNode is defined", async () => {
        const testNode = new ZoweUSSNode(
            "/u", vscode.TreeItemCollapsibleState.None, null, session, null, false, profileOne.name, undefined);
        testNode.contextValue = USS_SESSION_CONTEXT;

        expect(testNode.label).toBeDefined();
        expect(testNode.collapsibleState).toBeDefined();
        expect(testNode.label).toBeDefined();
        expect(testNode.getParent()).toBeDefined();
        expect(testNode.getSession()).toBeDefined();
    });

    /*************************************************************************************************************
     * Creates sample ZoweUSSNode list and checks that getChildren() returns the correct array
     *************************************************************************************************************/
    it("Testing that getChildren returns the correct Thenable<ZoweUSSNode[]>", async () => {
        // Creating a rootNode
        const rootNode = new ZoweUSSNode(
            "/u", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, undefined);
        rootNode.contextValue = USS_DIR_CONTEXT;
        rootNode.dirty = true;

        // Creating structure of files and directories
        const elementChildren = {};
        const sampleChildren: ZoweUSSNode[] = [
            new ZoweUSSNode("aDir", vscode.TreeItemCollapsibleState.Collapsed, rootNode, session, "/u", false, profileOne.name, undefined),
            new ZoweUSSNode("myFile.txt", vscode.TreeItemCollapsibleState.None, rootNode, session, "/u", false, profileOne.name, undefined),

        ];
        sampleChildren[1].command = { command: "zowe.uss.ZoweUSSNode.open", title: "Open", arguments: [sampleChildren[1]] };

        let rootChildren = await rootNode.getChildren();
        expect(rootChildren.length).toBe(2);
        expect(rootChildren[0].label).toBe("aDir");
        expect(rootChildren[1].label).toBe("myFile.txt");

        rootChildren[0].dirty = false;
        await rootChildren[0].getChildren();
        expect(rootChildren[0].children.length).toBe(0);

        // Check that error is thrown when label is blank
        const errorNode = new ZoweUSSNode(
            "", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, undefined);
        errorNode.dirty = true;
        expect(errorNode.getChildren()).rejects.toEqual(Error("Invalid node"));

        // Check that label is different when label contains a []
        const rootNode2 = new ZoweUSSNode(
            "root[test]", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, undefined);
        rootNode2.dirty = true;
        rootChildren = await rootNode2.getChildren();
    });

    /*************************************************************************************************************
     * Checks that the catch block is reached when an error is thrown
     *************************************************************************************************************/
    it("Checks that when bright.List. causes an error on the zowe call, " +
        "it throws an error and the catch block is reached", async () => {
            // Creating a rootNode
            const rootNode = new ZoweUSSNode(
                "toot", vscode.TreeItemCollapsibleState.Collapsed, null, session, "root", false, profileOne.name, undefined);
            rootNode.contextValue = USS_SESSION_CONTEXT;
            rootNode.fullPath = "Throw Error";
            rootNode.dirty = true;
            await rootNode.getChildren();
            expect(showErrorMessage.mock.calls.length).toEqual(1);
            expect(showErrorMessage.mock.calls[0][0]).toEqual("Retrieving response from uss-file-list Error: Throwing an error to check error handling for unit tests!");
        });

    /*************************************************************************************************************
     * Checks that returning an unsuccessful response results in an error being thrown and caught
     *************************************************************************************************************/
    it("Checks that when bright.List returns an unsuccessful response, " +
        "it throws an error and the catch block is reached", async () => {
            // Creating a rootNode
            const rootNode = new ZoweUSSNode(
                "toot", vscode.TreeItemCollapsibleState.Collapsed, null, session, "root", false, profileOne.name, undefined);
            rootNode.contextValue = USS_SESSION_CONTEXT;
            rootNode.dirty = true;
            const subNode = new ZoweUSSNode(
                "Response Fail", vscode.TreeItemCollapsibleState.Collapsed, rootNode, null, null, false, profileOne.name, undefined);
            subNode.fullPath = "THROW ERROR";
            subNode.dirty = true;
            await subNode.getChildren();
            expect(showErrorMessage.mock.calls.length).toEqual(1);
            expect(showErrorMessage.mock.calls[0][0]).toEqual("Retrieving response from uss-file-list Error: Throwing an error to check error handling for unit tests!");
        });

    /*************************************************************************************************************
     * Checks that passing a session node that is not dirty ignores the getChildren() method
     *************************************************************************************************************/
    it("Checks that passing a session node that is not dirty the getChildren() method is exited early", async () => {
        // Creating a rootNode
        const rootNode = new ZoweUSSNode(
            "root", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, undefined);
        rootNode.contextValue = USS_SESSION_CONTEXT;
        rootNode.dirty = false;
        expect(await rootNode.getChildren()).toEqual([]);
    });

    /*************************************************************************************************************
     * Checks that passing a session node with no hlq ignores the getChildren() method
     *************************************************************************************************************/
    it("Checks that passing a session node with no hlq the getChildren() method is exited early", async () => {
        // Creating a rootNode
        const rootNode = new ZoweUSSNode(
            "root", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, undefined);
        rootNode.contextValue = USS_SESSION_CONTEXT;
        expect(await rootNode.getChildren()).toEqual([]);
    });

    /*************************************************************************************************************
     * Checks that when getSession() is called on a memeber it returns the proper session
     *************************************************************************************************************/
    it("Checks that a child can reach its session properly", async () => {
        // Creating a rootNode
        const rootNode = new ZoweUSSNode(
            "root", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, undefined);
        rootNode.contextValue = USS_SESSION_CONTEXT;
        const subNode = new ZoweUSSNode(
            DS_PDS_CONTEXT, vscode.TreeItemCollapsibleState.Collapsed, rootNode, null, null, false, profileOne.name, undefined);
        const child = new ZoweUSSNode(
            "child", vscode.TreeItemCollapsibleState.None, subNode, null, null, false, profileOne.name, undefined);
        expect(child.getSession()).toBeDefined();
    });

    /*************************************************************************************************************
     * Checks that setBinary works
     *************************************************************************************************************/
    it("Checks that set Binary works", async () => {
        const rootNode = new ZoweUSSNode(
            FAVORITE_CONTEXT, vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, undefined);
        rootNode.contextValue = FAVORITE_CONTEXT;
        const subNode = new ZoweUSSNode(
            "binaryFile", vscode.TreeItemCollapsibleState.Collapsed, rootNode, null, null, true, profileOne.name, undefined);
        const child = new ZoweUSSNode(
            "child", vscode.TreeItemCollapsibleState.None, subNode, null, null, false, profileOne.name, undefined);

        child.setBinary(true);
        expect(child.contextValue).toEqual(DS_BINARY_FILE_CONTEXT);
        expect(JSON.stringify(child.iconPath)).toContain("document-binary.svg");
        child.setBinary(false);
        expect(child.contextValue).toEqual(DS_TEXT_FILE_CONTEXT);
        subNode.setBinary(true);
        expect(subNode.contextValue).toEqual(DS_BINARY_FILE_CONTEXT + FAV_SUFFIX);
        subNode.setBinary(false);
        expect(subNode.contextValue).toEqual(DS_TEXT_FILE_CONTEXT + FAV_SUFFIX);
    });

    /*************************************************************************************************************
     * Checks that labelRefresh works
     *************************************************************************************************************/
    it("Checks that labelRefresh subtley alters the label", async () => {
        const rootNode = new ZoweUSSNode(
            "gappy", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, undefined);
        expect(rootNode.label === "gappy");
        sharedUtils.labelRefresh(rootNode);
        expect(rootNode.label === "gappy ");
        sharedUtils.labelRefresh(rootNode);
        expect(rootNode.label === "gappy");
    });

    /*************************************************************************************************************
     * Checks that getEtag() returns a value
     *************************************************************************************************************/
    it("Checks that getEtag() returns a value", async () => {
        const rootNode = new ZoweUSSNode(
            "gappy", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, "123");
        expect(rootNode.getEtag() === "123");
    });

    /*************************************************************************************************************
     * Checks that setEtag() assigns a value
     *************************************************************************************************************/
    it("Checks that setEtag() assigns a value", async () => {
        const rootNode = new ZoweUSSNode(
            "gappy", vscode.TreeItemCollapsibleState.Collapsed, null, session, null, false, profileOne.name, "123");
        expect(rootNode.getEtag() === "123");
        rootNode.setEtag("ABC");
        expect(rootNode.getEtag() === "ABC");
    });
});
