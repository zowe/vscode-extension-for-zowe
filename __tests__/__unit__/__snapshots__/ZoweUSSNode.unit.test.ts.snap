// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Tests (Jest) Checks that the ZoweUSSNode structure matches the snapshot 1`] = `
ZoweUSSNode {
  "binary": false,
  "binaryFiles": Object {},
  "children": Array [],
  "collapsibleState": 0,
  "contextValue": "textFile",
  "dirty": true,
  "downloadedInternal": false,
  "downloadedTime": null,
  "etag": "",
  "fullPath": "",
  "iconPath": "Ref: 'document.svg'",
  "label": "testFile",
  "mParent": ZoweUSSNode {
    "binary": false,
    "binaryFiles": Object {},
    "children": Array [],
    "collapsibleState": 1,
    "contextValue": "directory",
    "dirty": true,
    "downloadedInternal": false,
    "downloadedTime": null,
    "etag": "",
    "fullPath": "",
    "iconPath": "Ref: 'folder.svg'",
    "label": "testDir",
    "mParent": ZoweUSSNode {
      "binary": false,
      "binaryFiles": Object {},
      "children": Array [],
      "collapsibleState": 1,
      "contextValue": "ussSession",
      "dirty": true,
      "downloadedInternal": false,
      "downloadedTime": null,
      "etag": "",
      "fullPath": "",
      "iconPath": "Ref: 'folder.svg'",
      "label": "root",
      "mParent": null,
      "mProfileName": "aProfile",
      "parentPath": null,
      "profile": Object {
        "failNotFound": false,
        "message": "",
        "name": "aProfile",
        "profile": Object {},
        "type": "zosmf",
      },
      "profileName": "",
      "session": Session {
        "ISession": Object {
          "hostname": "fake",
          "password": "fake",
          "protocol": "https",
          "type": "basic",
          "user": "fake",
        },
      },
      "shortLabel": "",
    },
    "mProfileName": "aProfile",
    "parentPath": null,
    "profile": Object {
      "failNotFound": false,
      "message": "",
      "name": "aProfile",
      "profile": Object {},
      "type": "zosmf",
    },
    "profileName": "",
    "session": null,
    "shortLabel": "",
  },
  "mProfileName": "aProfile",
  "parentPath": null,
  "profile": Object {
    "failNotFound": false,
    "message": "",
    "name": "aProfile",
    "profile": Object {},
    "type": "zosmf",
  },
  "profileName": "",
  "session": null,
  "shortLabel": "",
}
`;
